src/interp/interp.h

```diff
@@ -197,6 +202,11 @@ struct TypedValue {
           break;
       }
     }
+    // Nullable reftypes also have Nullref as a subtype
+    if ((type == Type::Funcref || type == Type::Hostref) &&
```

Comment:
I'm a little bit unsure about this. Basically it forces any value of type Funcref or Hostref which happens to be null, to be Nullref instead? I'm not exactly sure what `SetConcreteType` is used for in wabt though?

Reply:
Yes this is basically downcasting.   And nullref site below funcref and hostref in the type hierarchy.   So when the value stored is updated to store a null ref we also update the type of the TypedValue to reflect this.   I'll see if I can come up with better comment here.

Reply:
I'm still not sure why this is desirable though. Like, why do we prefer to force a downcast rather than letting the value keep its same type?

Reply:
(which I guess comes back to the fact that I don't know this codebase as well as a reviewer should... what is `TypedValue` actually used for?)

Reply:
Nevermind I think i was still looking at the outdated diff.